AppleCider/models/hyrax_models/HyraxBaselineCLS.py

```diff
@@ -0,0 +1,149 @@
+        super().__init__()
+
+        self.config = config
+        self._criterion = FocalLoss
+
+        model_config = config["model"]["BaselineCLS"]
+        #self.optimizer = torch.optim.AdamW(self.parameters(), lr=model_config["lr"], weight_decay=model_config["weight_decay"])
+        # Use AdamW eventually
+        #self.optimizer = torch.optim.Adam()
+
+        self.in_proj = nn.Linear(7, model_config['d_model'])
+        self.cls_tok = nn.Parameter(torch.zeros(1, 1, model_config['d_model']))
+
+        self.time2vec = Time2Vec(model_config['d_model'])
+
+        enc_layer = nn.TransformerEncoderLayer(
+            model_config['d_model'], model_config['n_heads'], model_config['d_model'] * 4,
+            model_config['dropout'], batch_first=True)
+        self.encoder = nn.TransformerEncoder(enc_layer, model_config['n_layers'])
+        self.norm = nn.LayerNorm(model_config['d_model'])
+        self.head = nn.Linear(model_config['d_model'], model_config['num_classes'])
+
+        self.classification = True if model_config['mode'] == 'photo' else False
+        if self.classification:
+            self.fc = nn.Linear(model_config['d_model'], model_config['num_classes'])
+
+    def forward(self, x, pad=None):
+        """
+        x: (B, L, 7)  - the raw event tensor from build_event_tensor
+            channels: [ dt, dt_prev, logf, logfe, one-hot-band(3) ]
+        pad_mask: (B, L) boolean
+        """
+        B, L, _ = x.shape
+
+        # project into model dim
+        h = self.in_proj(x)                     # (B, L, d_model)
+        # extract the *continuous* log1p dt feature
+        t = x[..., 0]
+
+        # compute the learned time embedding:
+        te = self.time2vec(t)
+
+        # add it:
+        hte = h + te                              # (B, L, d_model)
+        # prepend a learned CLS token:
+        tok = self.cls_tok.expand(B, -1, -1)      # (B,1,d_model)
+        hte = torch.cat([tok, hte], dim=1)        # (B, L+1, d_model)
+
+        # encode
+        z = self.encoder(hte, src_key_padding_mask=pad)  # (B, L+1, d_model)
+
+        output = self.norm(z[:, 0])  # (B, d_model )
+
+        if self.classification:
+            # classification from the CLS token
+            output = self.fc(output)  # (B, num_classes)
+
+        return output
+
+    def train_step(self, batch):
+        """
+        This function contains the logic for a single training step. i.e. the
+        contents of the inner loop of a ML training process.
+
+        Parameters
+        ----------
+        batch : tuple
+            A tuple containing the two values the loss function
+
+        Returns
+        -------
+        Current loss value : dict
+            Dictionary containing the loss value for the current batch.
+        """
+        data = batch[0]
+        labels = batch[1]
+        mask = batch[2]
+        self.optimizer.zero_grad()
+
+        decoded = self.forward(data, pad=mask)
```

Comment:
It would probably be worth considering both how the data is passed to `self.forward` during training and during inference. In both cases, the raw data will pass through `to_tensor`, however, for training, self.forward will be called by `train_step` and for inference, it will not. 

I would recommend that on this line, you call it like so: `decoded = self.forward(batch)`, and then unpack the tuple in self.forward.